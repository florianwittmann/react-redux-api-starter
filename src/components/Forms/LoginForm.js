import React from "react";
import { reduxForm } from "redux-form";
import { connect } from "react-redux";
import InputField from "./components/InputField";
import Alert from "./components/Alert";
import SubmitButton from "./components/SubmitButton";
import {
  isRequired,
  isEmail,
  validate as checkValidation
} from "./components/Validations";
import { loginUser } from "../../data/user/actions";
import { withRouter, Link } from "react-router-dom";

const LoginForm = props => {
  const onSubmit = ({ email, password }) => {
    props.loginUser({ email, password }, () => {
      props.history.push("/dashboard");
    });
  };
  return (
    <form onSubmit={props.handleSubmit(onSubmit)}>
      <InputField type="email" name="email" title="Email" icon="envelope" />
      <InputField
        name="password"
        title="Password"
        type="password"
        icon="lock"
      />
      <Alert errorMessage={props.errorMessage} />
      <SubmitButton inProgress={props.inProgress}>Login</SubmitButton>
      <Link to="/reset">Forgot your password?</Link>
      <hr />
      New customer? <Link to="/signup">Create new account</Link>

    </form>
  );
};

function validate(values) {
  return checkValidation(values, {
    email: [isRequired, isEmail],
    password: [isRequired]
  });
}

function mapStateToProps(state) {
  return { errorMessage: state.user.error, inProgress: state.user.inProgress };
}

const reduxFormLogin = reduxForm({ validate, form: "LoginForm" })(LoginForm);
export default withRouter(
  connect(mapStateToProps, { loginUser })(reduxFormLogin)
);
