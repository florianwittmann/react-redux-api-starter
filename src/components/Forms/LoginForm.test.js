import React from "react";
import { shallow } from "enzyme";
import { LoginForm } from "./LoginForm";

test("should render the LoginForm", () => {
  const wrapper = shallow(<LoginForm handleSubmit={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
