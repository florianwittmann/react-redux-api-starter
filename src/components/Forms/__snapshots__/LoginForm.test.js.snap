// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the LoginForm 1`] = `
<form>
  <InputField
    icon="envelope"
    name="email"
    title="Email"
    type="email"
  />
  <InputField
    icon="lock"
    name="password"
    title="Password"
    type="password"
  />
  <Component />
  <Component>
    Login
  </Component>
  <Link
    replace={false}
    to="/reset"
  >
    Forgot your password?
  </Link>
  <hr />
  New customer? 
  <Link
    replace={false}
    to="/signup"
  >
    Create new account
  </Link>
</form>
`;
